src/ol_infrastructure/components/services/vault.py

```diff
@@ -473,8 +472,8 @@ def __init__(
             backend=role_config.pki_intermediate_backend,
             allowed_domains=role_config.allowed_domains,
             allow_glob_domains=True,
-            key_type=role_config.key_type,
-            key_bits=role_config.key_bits,
+            key_type=VaultPKIKeyTypeBits.rsa.name,
+            key_bits=VaultPKIKeyTypeBits.rsa,
```

Comment:
This is hardcoding it to use RSA. You shoudl be able to just use `key_config.name, key_config.value`